### PR TITLE
chat clean up 

### PR DESCRIPTION
**Pull request by Dannygx21**
- chat submit on enter key
- chat render on user role
- user cant send message if role doesnt allow

### CODE REVIEW

File: src/Components/GhostChat.jsx [hunk 1]
```diff
@@ -19,12 +19,12 @@ const GhostChat = ({
...
-  const playerRole =
-  playerInfo
-    ?.find(player => player?.player_id === playerState?.player_id || null)
+  const player = playerInfo?.find(
+    (player) => player?.player_id === playerState?.player_id || null
+  );
 
   const handleMessageSubmit = (event) => {
-    if (newMessage.length > 0) {
+    if (newMessage.length > 0 && (player.role === 1 || player.role === 3)) {
```
Comment (thehimmat) [inline src/Components/GhostChat.jsx]: I would suggest && player.role % 2 === 1 to handle the new roles currently being added

File: src/Components/GhostChat.jsx [hunk 2]
```diff
@@ -39,23 +39,24 @@ const GhostChat = ({
...
-          } else {
-            return (
-              <UserBox key={i}>
-                <Username>{msg.username} </Username>
-                <Message>{msg.message} </Message>
-              </UserBox>
-            );
-          }
-        })}
+        {(player?.role === 1 || player?.role === 3) &&
```
Comment (thehimmat) [inline src/Components/GhostChat.jsx]: here as well (see comment above)

File: src/Components/LivingChat.jsx [hunk 3]
```diff
@@ -23,12 +23,12 @@ const LivingChat = ({
...
-  const playerRole =
-  playerInfo
-    ?.find(player => player?.player_id === playerState?.player_id || null)
+  const player = playerInfo?.find(
+    (player) => player?.player_id === playerState?.player_id || null
+  );
 
   const handleMessageSubmit = (event) => {
-    if (newMessage.length > 0) {
+    if (newMessage.length > 0 && (player.role === 0 || player.role === 2)) {
```
Comment (thehimmat) [inline src/Components/LivingChat.jsx]: similar to above - suggest changing to && player.role % 2 === 0

**Review by thehimmat — CHANGES_REQUESTED** [review]
all looks great. I would suggest changing some of the boolean conditions to handle the special roles that are being added soon. I added specific comments inline

File: src/Components/GhostChat.jsx (hunk 1 above)
Comment (Dannygx21) [reply src/Components/GhostChat.jsx]: on it

**Review by thehimmat — APPROVED** [review]
Approved!

**Pull request merged**